pyramid/tests/test_i18n.py

```diff
@@ -357,6 +357,36 @@ def test_add_same_domain_merge_true(self):
         inst.add(inst2)
         self.assertEqual(inst._catalog['a'], 'b')
 
+    def test_add_default_domain_replaces_plural_first_time(self):
+        # Create three empty message catalogs in the default domain
+        inst = self._getTargetClass()(None, domain='messages')
+        inst2 = self._getTargetClass()(None, domain='messages')
+        inst3 = self._getTargetClass()(None, domain='messages')
+        inst._catalog = {}
+        inst2._catalog = {}
+        inst3._catalog = {}
+
+        # The default plural scheme is the germanic one
+        self.assertEqual(inst.plural(0), 1)
+        self.assertEqual(inst.plural(1), 0)
+        self.assertEqual(inst.plural(2), 1)
+
+        # inst2 represents a message file that declares french plurals
+        inst2.plural = lambda n: n > 1
+        inst.add(inst2)
+        # that plural rule should now apply to inst
+        self.assertEqual(inst.plural(0), 0)
+        self.assertEqual(inst.plural(1), 0)
+        self.assertEqual(inst.plural(2), 1)
+
+        # We load a second message file with different plural rules
+        inst3.plural = lambda n: n > 0
+        inst.add(inst3)
+        # It doesn't override the previously loaded rule
+        self.assertEqual(inst.plural(0), 0)
+        self.assertEqual(inst.plural(1), 0)
+        self.assertEqual(inst.plural(2), 1)
```

Comment:
This is all in the context of a single language. 

The current behaviour is that the first plural wins, which for the default domain is the default pluraliser.

This change makes it so that for the default domain the second wins, which is the first user defined one.

There is an existing assumption that plurals within a language and domain pair are consistent which this patch doesn't affect